compiler/optimizer/abstractinterpreter/OMRIDTBuilder.hpp

```diff
@@ -0,0 +1,162 @@
+/*******************************************************************************
+ * Copyright IBM Corp. and others 2020
+ *
+ * This program and the accompanying materials are made available under
+ * the terms of the Eclipse Public License 2.0 which accompanies this
+ * distribution and is available at http://eclipse.org/legal/epl-2.0
+ * or the Apache License, Version 2.0 which accompanies this distribution
+ * and is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * This Source Code may also be made available under the following Secondary
+ * Licenses when the conditions for such availability set forth in the
+ * Eclipse Public License, v. 2.0 are satisfied: GNU General Public License,
+ * version 2 with the GNU Classpath Exception [1] and GNU General Public
+ * License, version 2 with the OpenJDK Assembly Exception [2].
+ *
+ * [1] https://www.gnu.org/software/classpath/license.html
+ * [2] https://openjdk.org/legal/assembly-exception.html
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0 OR GPL-2.0-only WITH Classpath-exception-2.0 OR GPL-2.0-only WITH OpenJDK-assembly-exception-1.0
+ *******************************************************************************/
+
+#ifndef OMR_IDT_BUILDER_INCL
+#define OMR_IDT_BUILDER_INCL
+
+/*
+ * The following #define and typedef must appear before any #includes in this file
+ */
+#ifndef OMR_IDT_BUILDER_CONNECTOR
+#define OMR_IDT_BUILDER_CONNECTOR
+namespace OMR { class IDTBuilder; }
+namespace OMR { typedef OMR::IDTBuilder IDTBuilderConnector; }
+#endif
+
+#include "il/ResolvedMethodSymbol.hpp"
+#include "env/Region.hpp"
+#include "compile/Compilation.hpp"
+#include "optimizer/Inliner.hpp"
+#include "optimizer/abstractinterpreter/AbsVisitor.hpp"
+#include "optimizer/abstractinterpreter/IDT.hpp"
+#include "optimizer/abstractinterpreter/IDTNode.hpp"
+#include "infra/vector.hpp"
```

Comment:
I kept `AbsVisitor.hpp` becasue it was used as base class, and `IDTNode.hpp` since the compiler threw out errors because many other classes which include `OMRIDTBuilder.hpp` also depend on the dependency in `IDTNode.hpp`. I also kept `vector.hpp` and includes `map` for clarity.

I just removed `ResolvedMethodSymbol.hpp` and `Region.hpp` since they were already in `IDTNode.hpp`, and I can do a explicity forward declaration for them if you want. Although removing `Inliner.hpp` won't cause any compilation error, I'm not sure which header file includes it, so I did a forward declaration for `class TR_InlinerBase`.